src/library/tabledelegates/keydelegate.cpp

```diff
@@ -0,0 +1,61 @@
+#include "library/tabledelegates/keydelegate.h"
+
+#include <qnamespace.h>
+
+#include <QPainter>
+#include <QStyle>
+#include <QTableView>
+
+#include "moc_keydelegate.cpp"
+#include "track/keyutils.h"
+#include "util/color/predefinedcolorpalettes.h"
+#include "util/color/rgbcolor.h"
+
+KeyDelegate::KeyDelegate(QTableView* pTableView)
+        : TableItemDelegate(pTableView) {
+}
+
+void KeyDelegate::paintItem(
+        QPainter* painter,
+        const QStyleOptionViewItem& option,
+        const QModelIndex& index) const {
+    paintItemBackground(painter, option, index);
+
+    QString keyText = index.data().toString();
+    mixxx::track::io::key::ChromaticKey key = KeyUtils::guessKeyFromText(keyText);
+    int openKeyNumber = KeyUtils::keyToOpenKeyNumber(key);
+
+    if (openKeyNumber != 0) {
+        const auto rgbColor =
+                mixxx::PredefinedColorPalettes::kDefaultKeyColorPalette.at(
+                        openKeyNumber - 1); // Open Key numbers start from 1
+        const auto qcolor = mixxx::RgbColor::toQColor(rgbColor);
+
+        // Draw the colored rectangle next to the key label
+        painter->fillRect(
+                option.rect.x() + 2,
+                option.rect.y() + 2,
+                4, // width
+                option.rect.height() - 4,
+                qcolor);
+    }
+
+    const int rectWidth = 10; // 2px left padding + 4px width + 4px right padding
+
+    QString elidedText = option.fontMetrics.elidedText(
+            index.data().toString(),
+            Qt::ElideLeft,
+            columnWidth(index) - rectWidth);
+
```

Comment:
Like in LocationDelegate we need to get the highlight color from the palette in order to paint the selected key text correctly:
https://github.com/mixxxdj/mixxx/blob/a48562ad96021bd9571c8fbc8d331291bc30c1fc/src/library/tabledelegates/locationdelegate.cpp#L11-L23

Reply:
Thank you! Just fixed this.